core/src/main/java/org/apache/struts2/interceptor/csp/DefaultCspSettings.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.struts2.interceptor.csp;
+
+import com.opensymphony.xwork2.ActionContext;
+
+import javax.servlet.http.HttpServletResponse;
+import java.security.SecureRandom;
+import java.util.Base64;
+import java.util.Map;
+
+
+public class DefaultCspSettings implements CspSettings {
+
+    private static final int NONCE_LENGTH = 18;
+    private final SecureRandom sRand  = new SecureRandom();
+    private String reportUri = "cspviolation";
+    private boolean reporting = true;
```

Comment:
these two settings imply that CSP will be enabled in report mode by default, which is the right thing to do! the default value for the `reportUri` is too specific, though, let's maybe rename it to be something along the lines of `/csp-reports`